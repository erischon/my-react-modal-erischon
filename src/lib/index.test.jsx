import React from "react";
import renderer from "react-test-renderer";

import { describe, expect, it } from "vitest";

import MyModal from "./index";

const modalProps = {
  title: "Modal Title",
  message: "Modal message",
  actionA: {
    title: "Action A Title",
    action: () => {},
  },
  actionB: {
    title: "Action B Title",
    action: () => {},
  },
  isOpen: () => {},
};

describe("MyModal Component", () => {
  it("renders correctly", () => {
    const component = renderer.create(<MyModal modalProps={modalProps} />);

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
