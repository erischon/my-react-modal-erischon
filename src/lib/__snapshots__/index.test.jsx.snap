// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyModal Component > renders correctly 1`] = `
<div
  className="z-50 transition duration-300 bg-zinc-600 bg-opacity-80 flex justify-center items-center overflow-x-hidden overflow-y-auto fixed inset-0"
>
  <div
    className="bg-zinc-100 w-11/12 md:max-w-md mx-auto rounded-md shadow-lg z-50 overflow-y-auto"
  >
    <div
      className="flex justify-between items-start p-4 gap-4"
    >
      <h2
        className="text-xl font-bold"
      >
        Modal Title
      </h2>
      <div
        className="flex justify-center items-center cursor-pointer"
      >
        <svg
          className="text-xl font-bold"
          fill="currentColor"
          height="1em"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": undefined,
            }
          }
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 0 0 203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
          />
        </svg>
      </div>
    </div>
    <div
      className="py-4 text-left px-4"
    >
      Modal message
    </div>
    <div
      className="flex gap-2 items-center w-100 p-4"
    >
      <button
        className="bg-red-700 rounded-md px-3 py-1 text-white font-semibold"
        onClick={[Function]}
      >
        Action A Title
      </button>
      <button
        className="bg-red-700 rounded-md px-3 py-1 text-white font-semibold"
        onClick={[Function]}
      >
        Action B Title
      </button>
    </div>
  </div>
</div>
`;
